pygls/protocol.py

```diff
@@ -427,6 +437,17 @@ def connection_made(self, transport: asyncio.BaseTransport):
         self.transport = transport
 
     def data_received(self, data: bytes):
+        try:
+            self._data_received(data)
+        except Exception:
+            logger.error(traceback.format_exc())
```

Comment:
I know that this pattern is present in other places, but instead of:
```py
            logger.error(traceback.format_exc())
```
we can use logging ability to log the currently handled exception with something like this:

```py
            logger.error("Error receiving data", exc_info=True)
```

Reply:
Updated in latest force push